<!-- -->
<launch>

  <arg name="method_type" default="0" /> <!-- pcl_generic=0, pcl_anh=1, pcl_anh_gpu=2, pcl_openmp=3 -->
  <arg name="use_odom" default="true" />
  <arg name="use_imu" default="false" />
  <arg name="imu_upside_down" default="false" />
  <arg name="imu_topic" default="/imu_raw" />
  <arg name="incremental_voxel_update" default="false" />

  <!-- ceres-optimizer -->
  <arg name="optimize_num_every_scans" default="10" />
  <arg name="optimize_iter_num" default="60" />
  <arg name="optimize_var_anchor" default="0.8" />
  <arg name="optimize_var_odom_t" default="0.2" />
  <arg name="optimize_var_odom_q" default="0.02" />
  
  <!-- rosrun lidar_localizer ceres_ndt_mapping  -->
  <!-- <node pkg="lidar_localizer" type="queue_counter" name="queue_counter" output="screen"/> -->
  <!-- nmea2tfpose -->
  <include file="$(find gnss_localizer)/launch/nmea2tfpose.launch"/>
  <node pkg="lidar_localizer" type="ceres_ndt_mapping" name="ceres_ndt_mapping" output="screen">
    <param name="map_init_pos_file" value="/autoware/workspace/data/map/hengtong_1115_map_init.txt" />
    <param name="config_time_threshold" value="0.10" />
    <param name="method_type" value="$(arg method_type)" />
    <param name="use_imu" value="$(arg use_imu)" />
    <param name="use_odom" value="$(arg use_odom)" />
    <param name="imu_upside_down" value="$(arg imu_upside_down)" />
    <param name="imu_topic" value="$(arg imu_topic)" />
    <param name="incremental_voxel_update" value="$(arg incremental_voxel_update)" />

    <!-- back_end optimizer params -->
    <param name="optimize_num_every_scans" value="$(arg optimize_num_every_scans)" />
    <param name="optimize_iter_num" value="$(arg optimize_iter_num)" />
    <param name="optimize_var_anchor" value="$(arg optimize_var_anchor)" />
    <param name="optimize_var_odom_t" value="$(arg optimize_var_odom_t)" />
    <param name="optimize_var_odom_q" value="$(arg optimize_var_odom_q)" />
  </node>

</launch>
